pymc3/distributions/distribution.py

```diff
@@ -355,14 +373,19 @@ def _draw_value(param, point=None, givens=None):
         if point and hasattr(param, 'model') and param.name in point:
             return point[param.name]
         elif hasattr(param, 'random') and param.random is not None:
-            return param.random(point=point, size=None)
+            return param.random(point=point, size=size)
+        elif hasattr(param, 'distribution') and hasattr(param.distribution, 'random') and param.distribution.random is not None:
+            return param.distribution.random(point=point, size=size)
         else:
             if givens:
                 variables, values = list(zip(*givens))
             else:
                 variables = values = []
             func = _compile_theano_function(param, variables)
-            return func(*values)
+            if size is None:
+                return func(*values)
+            else:
+                return np.array([func(*value) for value in zip(*values)])
```

Comment:
The `size` seems to only be imposed to `param`'s with a `random` method, and we hope the content of `values` to be the right `size` in the end. Shouldn't there be some enforcement of the `size`, for the `numbers.Number`, `np.ndarray`, `tt.TensorConstant`, `tt.sharedvar.SharedVariable` and `tt.TensorVariable` in point cases for us to be sure that `values` will in fact have the desired output size?

Reply:
I am relying here on `theano` catching those sorts of errors, and giving more informative errors than I could.  I am running this on a few different models to make sure it gives reasonable results, but so far those sorts of inputs get broadcast in a sensible manner.